File: packages/java/anip-core/src/main/java/dev/anip/core/CapabilityOutput.java
package dev.anip.core;

import java.util.List;

/**
 * Describes the output of a capability.
 */
public class CapabilityOutput {

    private final String type;
    private final List<String> fields;

    public CapabilityOutput(String type, List<String> fields) {
        this.type = type;
        this.fields = fields;
    }

    public String getType() {
        return type;
    }

    public List<String> getFields() {
        return fields;
    }
}

File: packages/csharp/src/Anip.Core/CapabilityOutput.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class CapabilityOutput
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "";

    [JsonPropertyName("fields")]
    public List<string> Fields { get; set; } = new();
}
